opentracing-impl/src/main/java/io/opentracing/propagation/TextMapImpl.java

```diff
@@ -16,7 +16,7 @@
 import java.util.Iterator;
 import java.util.Map;
 
-public final class TextMapImpl implements TextMapWriter, TextMapReader {
+public final class TextMapImpl implements TextMap {
```

Comment:
> In other languages we haven't provided text map implementations as part of the core API. 

We do have `TextMapCarrier` in [Go](https://github.com/opentracing/opentracing-go/blob/master/propagation.go#L110). That one is actually R/W, but it's ok because in Go the accepted carrier interfaces are separated to R and W.